x-pack/plugins/security/server/authentication/providers/basic.ts

```diff
@@ -0,0 +1,213 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable max-classes-per-file */
+
+import { KibanaRequest } from '../../../../../../src/core/server';
+import { canRedirectRequest } from '../can_redirect_request';
+import { AuthenticationResult } from '../authentication_result';
+import { DeauthenticationResult } from '../deauthentication_result';
+import { BaseAuthenticationProvider } from './base';
+
+/**
+ * Utility class that knows how to decorate request with proper Basic authentication headers.
+ */
+export class BasicCredentials {
+  /**
+   * Takes provided `username` and `password`, transforms them into proper `Basic ***` authorization
+   * header and decorates passed request with it.
+   * @param request Request instance.
+   * @param username User name.
+   * @param password User password.
+   */
+  public static decorateRequest<T extends KibanaRequest>(
+    request: T,
+    username: string,
+    password: string
+  ) {
+    const typeOfRequest = typeof request;
+    if (!request || typeOfRequest !== 'object') {
+      throw new Error('Request should be a valid object.');
+    }
+
+    if (!username || typeof username !== 'string') {
+      throw new Error('Username should be a valid non-empty string.');
+    }
+
+    if (!password || typeof password !== 'string') {
+      throw new Error('Password should be a valid non-empty string.');
+    }
+
+    const basicCredentials = Buffer.from(`${username}:${password}`).toString('base64');
+    request.headers.authorization = `Basic ${basicCredentials}`;
+    return request;
+  }
+}
+
+/**
+ * Describes the parameters that are required by the provider to process the initial login request.
+ */
+interface ProviderLoginAttempt {
+  username: string;
+  password: string;
+}
+
+/**
+ * The state supported by the provider.
+ */
+interface ProviderState {
+  /**
+   * Content of the HTTP authorization header (`Basic base-64-of-username:password`) that is based
+   * on user credentials used at login time and that should be provided with every request to the
+   * Elasticsearch on behalf of the authenticated user.
+   */
+  authorization?: string;
+}
+
+/**
+ * Provider that supports request authentication via Basic HTTP Authentication.
+ */
+export class BasicAuthenticationProvider extends BaseAuthenticationProvider {
+  /**
+   * Performs initial login request using username and password.
+   * @param request Request instance.
+   * @param attempt User credentials.
+   * @param [state] Optional state object associated with the provider.
+   */
+  public async login(
+    request: KibanaRequest,
+    attempt: ProviderLoginAttempt,
+    state?: ProviderState | null
+  ) {
+    this.logger.debug('Trying to perform a login.');
+
+    if (!attempt || !attempt.username || !attempt.password) {
+      this.logger.debug('Username and/or password not provided.');
+      return AuthenticationResult.notHandled();
```

Comment:
Yeah, you're right, it probably makes sense to just remove this check entirely. We shouldn't do such checks for every argument we expect, especially since they are defined as required arguments.